ui/src/app/store/machine/types/base.ts

```diff
@@ -111,11 +108,22 @@ export type MachineDetails = BaseMachine &
     storage_layout_issues: string[];
     storage_test_status: TestStatus;
     supported_filesystems: SupportedFilesystem[];
-    sync_interval: Seconds;
     swap_size: number | null;
     testing_start_time: string;
   };
 
+type HardwareSyncFields =
+  | {
+      enable_hw_sync: false;
+    }
+  | {
+      enable_hw_sync: true;
+      last_sync: string;
+      next_sync: string;
+      is_sync_healthy: boolean;
+      sync_interval: Seconds;
+    };
```

Comment:
Interesting. This isn't how the websocket API *normally* works - we usually get back everything, with `null` returned for the values that are not applicable.

On one hand it makes sense to only get back the data that's relevant, and with this granularity we can be more confident in the possible value combinations for a machine. On the other hand this opens up the `Machine` type to take many different shapes, where each new conditional like this multiplies the possible `Machine` shapes by two. I think that could quickly become problematic, or at least be a hassle having to coerce the type in multiple ways in multiple places. I'm leaning toward asking if we could update the API to return `null`, if only for simplicity and consistency's sake.

What do you think @petermakowski @huwshimi?

Reply:
Actually I've changed my mind, having to null check each of these values even when you *know* they're not null sounds like more of a hassle. Forget everything I just said hahaha